src/tables/math.rs

```diff
@@ -0,0 +1,1062 @@
+/// in the correctionHeights array, the first kerning value (index 0) is used. For a height that is
+/// greater than or equal to the last entry in the correctionHeights array, the last entry is used.
+#[allow(missing_debug_implementations)]
+#[derive(Clone)]
+pub struct MathKern<'a> {
+    data: &'a [u8],
+    correction_height: LazyArray16<'a, RawMathValueRecord>,
+    kern_values: LazyArray16<'a, RawMathValueRecord>,
+}
+
+impl<'a> MathKern<'a> {
+    /// Number of heights as split points.
+    pub fn height_count(&self) -> u16 { self.correction_height.len() }
+    /// Number of kern values. It is always greater than number of heights by one.
+    pub fn kern_count(&self) -> u16 { self.kern_values.len() }
+    /// Get the height value at some specific index.
+    pub fn get_height(&self, index: u16) -> Option<MathValueRecord> {
+        Some(self.correction_height.get(index)?.get(self.data))
+    }
+    /// Get the kern value at some specific index.
+    pub fn get_kern_value(&self, index: u16) -> Option<MathValueRecord> {
+        Some(self.kern_values.get(index)?.get(self.data))
+    }
+}
+
+impl<'a> FromSlice<'a> for MathKern<'a> {
+    fn parse(data: &'a [u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        let count = s.read::<u16>()?;
+        let correction_height = s.read_array16(count)?;
+        let kern_values = s.read_array16(count + 1)?;
+        Some(MathKern { data, correction_height, kern_values })
+    }
+}
+
+/// The MathVariants table solves the following problem: given a particular default glyph shape and
+/// a certain width or height, find a variant shape glyph (or construct created by putting several
+/// glyphs together) that has the required measurement. This functionality is needed for growing
+/// the parentheses to match the height of the expression within, growing the radical sign to match
+/// the height of the expression under the radical, stretching accents like tilde when they are put
+/// over several characters, for stretching arrows, horizontal curly braces, and so forth.
+#[allow(missing_debug_implementations)]
+#[derive(Clone)]
+pub struct MathVariants<'a> {
+    /// Minimum overlap of connecting glyphs during glyph construction, in design units.
+    ///
+    /// `min_connector_overlap` defines by how much two glyphs need to overlap with each other when
+    /// used to construct a larger shape. Each glyph to be used as a building block in constructing
+    /// extended shapes will have a straight part at either or both ends. This connector part is
+    /// used to connect that glyph to other glyphs in the assembly. These connectors need to overlap
+    /// to compensate for rounding errors and hinting corrections at a lower resolution. The
+    /// `min_connector_overlap` value tells how much overlap is necessary for this particular font.
+    pub min_connector_overlap: u16,
+    /// Offset to Coverage table, from the beginning of the MathVariants table.
```

Comment:
It's not an offset anymore. We're already resolved it.

Reply:
Changed the documentation.

It is a bit repetitive in its current form. I plan to add a `CoverageTable` abstraction to group the `Coverage` and the array, because this way the data should appear better organised, and we can add some higher-level glyph-based API for looking up those arrays.

Reply:
I introduced a new type `CoverageMapping` for combining the `Coverage` table and the array for values. For the abstraction to work, I had to add a public trait `Array` in module `parser`, with two associated types `Index` and `Value`, and three public methods `len`, `is_empty`, and `get`. Unfortunately this is a breaking change (because these three are not inherent methods any more, and cannot be accessed without a `use parser::Array;`). Personally I think it is good to have a universal interface for all the arrays, but it is really up to you to decide. I hope this change is acceptable to you.

Relevant code:
https://github.com/ruifengx/ttf-parser/blob/math-table/src/parser.rs#L243-L255
https://github.com/ruifengx/ttf-parser/blob/math-table/src/tables/math.rs#L707-L715